web/src/browser_history.ts

```diff
@@ -168,6 +168,11 @@ export const state_data_schema = z.object({
 
 type StateData = z.infer<typeof state_data_schema>;
 
+export function current_scroll_offset(): number | undefined {
+    const current_state = state_data_schema.nullable().parse(window.history.state);
+    return current_state?.narrow_offset;
+}
```

Comment:
Why do we expect this to be accurate? I didn't think we updated `narrow_offset` when scrolling, just when switching views, so this should be wrong if we currently have the combined feed open? Not sure.

Reply:
Might be one just needs to check if it's the current version before calling this?

Reply:
> Why do we expect this to be accurate? 


`narrow_offset` is updated on every `message_selected.zulip` event via `narrow_history.save_narrow_state`. So, should be accurate.